Add ubuntu:20.04 to test matrix

mbargull opened the pull request:
Test for `SCMP_ACT_KILL_PROCESS` issue described at gh-232.

mbargull commented on Makefile.am:
```diff
@@ -61,7 +61,7 @@ ci:
 	git diff --exit-code
 # TODO: make sure ./vendor is synced with ./vendor.sh
-	$(MAKE) lint
+#	$(MAKE) lint
```
clang-tidy complains on Ubuntu 20.04 about `memset`, `strncpy`, etc. vs `memset_s`, `strncpy`, etc. (C11).

mbargull commented on Dockerfile.tests:
```diff
@@ -2,7 +2,7 @@ ARG LIBSLIRP_COMMIT=v4.3.1
 
 FROM ubuntu:${UBUNTU_VERSION} AS build
-RUN apt update && apt install -y automake autotools-dev make gcc libglib2.0-dev libcap-dev libseccomp-dev git ninja-build python3-pip
+RUN apt update && DEBIAN_FRONTEND=noninteractive apt install -y automake autotools-dev make gcc libglib2.0-dev libcap-dev libseccomp-dev git ninja-build python3-pip
```
`tzdata` install requires interaction on Ubuntu 20.04.

AkihiroSuda commented on .github/workflows/main.yaml:
```diff
@@ -6,11 +6,13 @@ jobs:
     strategy:
       matrix:
         libslirp_commit: [master, v4.3.1, v4.2.0, v4.1.0]
```
Can we rather use Ubuntu 20.04 instance of GitHub Action so that we can use real Ubuntu 20.04 kernel

mbargull replied:
Makes sense to test that too, right. (Although the issue from gh-232 should be just userspace, right?)
My intent with this PR was just to show that it fails without gh-232 and works if it's applied. Hence, it's rather temporary and in a non-merge-able state.
However, the tests fail at another check, currently:
```
+ slirp_pid=2071
+ wait_for_network_device 2045 tun11
+ COUNTER=0
+ '[' 0 -lt 40 ']'
+ nsenter --preserve-credentials -U -n --target=2045 ip addr show tun11
+ slirp4netns -c 2045 --cidr 10.0.135.128/25 tun11
Device "tun11" does not exist.
+ sleep 0.5
sent tapfd=5 for tun11
received tapfd=5
+ let COUNTER=COUNTER+1
+ '[' 1 -lt 40 ']'
+ nsenter --preserve-credentials -U -n --target=2045 ip addr show tun11
2: tun11: <BROADCAST,UP,LOWER_UP> mtu 1500 qdisc fq_codel state UNKNOWN group default qlen 1000
    link/ether 9e:69:47:51:c9:3c brd ff:ff:ff:ff:ff:ff
    inet 10.0.135.228/25 brd 10.0.135.255 scope global tun11
       valid_lft forever preferred_lft forever
    inet6 fe80::9c69:47ff:fe51:c93c/64 scope link tentative 
       valid_lft forever preferred_lft forever
+ break
+ trap cleanup EXIT
++ nsenter --preserve-credentials -U -n --target=2045 ip -json a show dev tun11
++ jq -r '.[1].addr_info[0].local'
+ ip=null
+ [[ null = 10.0.135.228 ]]
+ cleanup
+ kill -9 2045 2071
FAIL tests/test-slirp4netns-cidr.sh (exit status: 1)
```

mbargull commented:
There doesn't appear to be any test that runs with `--enable-seccomp`.

mbargull commented on tests/test-slirp4netns-configure.sh:
```diff
@@ -8,7 +8,7 @@ child=$!
 wait_for_network_namespace $child
 
-slirp4netns -c $child tun11 &
+slirp4netns --enable-seccomp -c $child tun11 &
```
Hmm.. This was supposed to fail but it didn't. I'll have to do some more testing..

mbargull commented:
This is not able to test for the issue from gh-232 because we'd need an older kernel on the CI to test this.

----
That said, let me summarize what would be needed for using Ubuntu 20.04 on the CI:
1. Apt installs in `Dockerfile.tests` need `DEBIAN_FRONTEND=noninteractive apt install ...` because installing `tzdata` asks for user input.
2. `nmap` does not include `ncat` in Ubuntu 20.04 but "suggests" a separate package for this.
   I added `$( [ "${UBUNTU_VERSION}" = 20.04 ] && echo ncat )` to the `apt install` package list for this.
3. `clang-tidy` complains on Ubuntu 20.04 about `memset`, `strncpy`, etc. vs `memset_s`, `strncpy_s`, etc. (C11).
4. In `tests/test-slirp4netns-cidr.sh` I changed `jq -r .[1].addr_info[0].local` to `jq -r .[].addr_info[0].local`.
   (NB: I didn't look at the actual `ip` output, hence can't tell what the exact cause for the need for this change is.)
5. In `tests/test-slirp4netns-disable-dns.sh` I changed `grep "Connection timed out"` to `grep 'Connection timed out\|TIMEOUT'`.
   The error message of `ncat` changed, apparently.

Additionally, a test file `tests/test-slirp4netns-seccomp.sh` is added to test `--enable-seccomp`, which doesn't appear to be covered by any other test.
   The command that is run is `slirp4netns -c --enable-seccomp --userns-path=/proc/$child/ns/user $child tun11`.
   (The `--enable-seccomp --userns-path` combination is what fails if run on an older kernel if `slirp4nets` is compiled against `libseccomp>=2.4`.)
